### DISCUSSION TITLE
DatePicker in combination with dayjs.locale

### DISCUSSION THREAD

**Discussion by hensansi** [mantinedev/mantine#3742] [Q&A]
I'm not sure if it is a bug or if the configuration of the DatePicker/DateProvider changed in such a way that I'm not able to configure it properly.

When we set the DatePicker in combination with the days.locale, the header of the weekdays change. This is a regression/difference that I found when migrating from mantine 5 to mantine 6 and I didn't see anything on the documentation regarding it. Am I doing something wrong here?

CodeSandbox:
https://codesandbox.io/s/dates-locale-28jcn3?file=/src/App.tsx (uncomment the `dayjs.locale("en-gb")` to see the days of the week shifting)

In the images see how for example 10th of April is Monday on one calendar and Tuesday on the other.

With dayjs.locale:
![Screenshot 2023-03-10 at 13-47-20 CodeSandbox](https://user-images.githubusercontent.com/241124/224319897-27037e04-95d1-40b5-bd38-4169722ffbe8.png)

Without:
![Screenshot 2023-03-10 at 13-47-00 CodeSandbox](https://user-images.githubusercontent.com/241124/224319907-f6ced75d-5beb-47d9-be0d-6abaffda7ff8.png)

**Comment by rtivital** [comment]
You can change locale on DatesProvider or in component props. Changing locale with `dayjs.locale()` function is not supported

**Reply by hensansi** [reply]
Sorry, I was away from the computer during the weekend.

 I think I didn't explain the problem well, dayjs.locale() is set globally so we can display dates properly all around. This example is just showing the issue and in this case the Datepicker is picking up something from it too and then displaying the dates of the week incorrectly. This wasn't the case with Mantine 5.

**Reply by hensansi** [reply]
I guess I can for the moment reset the locale on the component level and that solves the issue but the change on Mantine 6 might introduce bugs on other people's code,
https://codesandbox.io/s/dates-locale-forked-w5yfzd?file=/src/App.tsx

**Reply by hensansi** [reply]
@rtivital ~~I was looking a bit further into the code and I found a a fix, although as I said it feels like it is something that needs to be resolved in the library. I can try to look further into the code and see if I can find a fix even though I have no idea where these props are coming from.~~

 ~~Here is a codesandbox with a solution:
https://codesandbox.io/s/dates-locale-forked-rgp8h3?file=/src/App.tsx~~

Ignore the solution, it was due to the fact that the component with the reset was before the solutions, once I change the order it doesn't work

**Reply by hensansi** [reply]
After some trial and error I can see where the problem stems from, if the locale data weekStart is different from 0 or undefined, mantine shows something wrong.

To hotfix this, if a locale has a different weekStart than 0 or undefined and it is set at global level we just need to set it to 0 like:

```
import updateLocale from "dayjs/plugin/updateLocale";
dayjs.extend(updateLocale);

(...)
dayjs.locale("en-gb");
// changing to other that isn't 0 or undefined breaks the week header
dayjs.updateLocale("en-gb", {
  weekStart: 0 
}); 
```
https://codesandbox.io/s/dates-locale-forked-xgh1is

@rtivital I think that this is a bug although I don't know exactly how to proceed here.

**Reply by rtivital** [reply]
I do not know either

**Reply by hensansi** [reply]
I will look into the issue this week, maybe I can contribute with something if I find the cause.
